sbt-bridge/test/xsbti/compile/TestCompileProgress.scala

```diff
@@ -0,0 +1,20 @@
+package xsbti.compile
+
+import org.junit.Assert.*
+
+import scala.collection.mutable
+
+class TestCompileProgress extends CompileProgress {
+
+  val startUnitCalls: mutable.Buffer[(String, String)] = mutable.Buffer.empty[(String, String)]
+
+  override def startUnit(phase: String, unitPath: String): Unit =
+    startUnitCalls += ((phase, unitPath))
+
+  // TODO: we should also call advance with calculated current / total
+  //  Beware: scala3-compiler uses compiler-interface 1.3.5 which is binary incompatible
+  //  with the one used un scala3-sbt-bridge (1.4.5)
+  //  it has different signature def advance(current: Int, total: Int): Boolean
```

Comment:
Maybe it's best to wait until we've upgraded to the latest interface before proceeding further here (see https://github.com/lampepfl/dotty/issues/10816, it seems this is currently blocked on bloop upgrading zinc /cc @tgodzik)

Reply:
Yeah, trying to get to that, but haven't managed to do it yet. We should also upgrade zinc in Gradle after the Scala 3 support is merged.